File: ctms-frontend/src/model/container_truck.ts
export type ContainerTruckResponse = {
  containerTrucks: ContainerTruck[];
  totalPage: number;
};

export type ContainerTruck = {
  truckId?: string;
  licensePlate: string;
  manufacturer: string;
  registrationDate: string;
  capacity: number;
  isActive: boolean;
  driver?: Driver;
  attach?: string;
  containerStatus?: string;
  orderId?: string;
};

export type Driver = {
  userId: string;
  username?: string;
  firstName?: string;
  lastName?: string;
  address?: string;
  phone?: string;
  email?: string;
  personalId?: string;
  image?: string;
  birthDate?: string;
  userNumber?: string;
};

File: ctms-backend/ctms/src/main/java/vn/fpt/edu/ctms/model/ContainerTruck.java
package vn.fpt.edu.ctms.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Date;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "container_truck")
@Entity
@Builder
public class ContainerTruck {
    @Id
    @Column(name = "truck_id")
    private String truckId;

    @Column(name = "license_plate")
    private String licensePlate;

    private String manufacturer;

    @Column(name = "registration_date")
    private LocalDate registrationDate;

    private Float capacity;

    @Column(name = "is_active")
    private Boolean isActive;

    private String attach;
    @Column(name="container_status")
    @Enumerated(EnumType.STRING)
    private ContainerStatus containerStatus;

    @Column(name = "in_use")
    private Boolean inUse;
    @OneToOne
    @JoinColumn(name = "driver", referencedColumnName = "user_id")
    private User driver;


    @PrePersist
    public void generateTruckId() {
        this.truckId = UUID.randomUUID().toString();
    }
}
